### PR TITLE
#[7752] fix warning about array type inference

### LINKED ISSUE #7752: CI failure on development
### Bug Description
See last [run](https://github.com/hzi-braunschweig/SORMAS-Project/runs/4899566809?check_suite_focus=true#step:7:882) on `development`.

### PR DESCRIPTION
**Pull request by JonasCir**
Fixes #7752 
Interestingly, I could not reproduce the failure locally, but if IIRC I saw this error last week in one of my branches.

IDE showed some warnings that missing array tape might cause trouble, so refactored the call site a little bit.
While I was at it, I also fixed some low hanging warnings.

Cheers

### CODE REVIEW

File: sormas-backend/src/main/java/de/symeda/sormas/backend/sample/SampleFacadeEjb.java [hunk 1]
```diff
@@ -979,18 +979,18 @@ private void onSampleChanged(SampleDto existingSample, Sample newSample, boolean
 
 	private void handleAssotiatedObjectChanges(Sample newSample, boolean syncShares) {
 		if (newSample.getAssociatedCase() != null) {
-			caseFacade.onCaseChanged(CaseFacadeEjbLocal.toDto(newSample.getAssociatedCase()), newSample.getAssociatedCase(), syncShares);
+			caseFacade.onCaseChanged(CaseFacadeEjb.toDto(newSample.getAssociatedCase()), newSample.getAssociatedCase(), syncShares);
```
Comment (MateStrysewske) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/sample/SampleFacadeEjb.java:982]: Afaik there's a reason we're using the local bean here; @MartinWahnschaffe can you share some insight?

Comment (JonasCir) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/sample/SampleFacadeEjb.java:982]: I think the warning originated from the usage of a static function on the extended class.

Comment (MartinWahnschaffe) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/sample/SampleFacadeEjb.java:982]: What you have in mind is that when using an instance of an EJB in the backend, you have to use the "local" instance.

The toDto method is a static method though, so there is no difference between using the static method via the class name of the ejb or any inherited class. As Jonas points out, the system will give a warning that says we should use the class that acutally defines the static method.

TL;DR: The refactoring is good the way it is.

Comment (JonasCir) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/sample/SampleFacadeEjb.java:982]: :+1:

**Review by MateStrysewske — APPROVED** [review]
